lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -1227,6 +1227,30 @@ LogicalResult XorOp::canonicalize(XorOp op, PatternRewriter &rewriter) {
   auto size = inputs.size();
   assert(size > 1 && "expected 2 or more operands");
 
+  // Check De Morgan's laws
+  Value sub;
+  if (matchPattern(static_cast<Operation *>(op), m_Complement(m_Any(&sub)))) {
+    auto getNegations = [&](auto term) {
+      SmallVector<Value, 2> negations;
+      auto operands = term->getOperands();
+      negations.reserve(operands.size());
+      llvm::transform(operands, std::back_inserter(negations), [&](Value x) {
+        return createOrFoldNot(op.getLoc(), x, rewriter);
+      });
+      return negations;
+    };
```

Comment:
The `createOrFoldNot` function probably needs to check if the operand is already a negation, and if that's the case, strip away the existing negation. Otherwise this might end up creating long chains of inversions before the canonicalizer has a chance to clean them up.

Reply:
Double negation elimination is implemented in the folder of `XorOp`,  so I would assume that chains of inversions are properly eliminated at creations.